packages/helpers-gatsby/src/Tags.js

```diff
@@ -36,37 +36,54 @@ const Tags = (props) => {
         visibility: props.visibility,
     }
 
+    let keyIndex = 0
+    const generateKey = (pre) => {
+        keyIndex = keyIndex + 1
+        return `${pre}_${keyIndex}`
+    }
+
     if (props.separator) {
-        opts.separator = React.isValidElement(props.separator) ? props.separator :
-            <><span className={props.separatorClasses}>{props.separator}</span></>
+        if (React.isValidElement(props.separator)) {
+            opts.separator = props.separator
+        } else {
+            Object.defineProperty(opts, 'separator', {
+                get() {
+                    return (
+                        <span className={props.separatorClasses} key={generateKey('separator')}>
+                            {props.separator}
+                        </span>
+                    )
+                },
+            })
+        }
     }
 
     if (props.prefix) {
         opts.prefix = React.isValidElement(props.prefix) ? props.prefix :
-            <><span className={props.prefixClasses}>{props.prefix}</span></>
+            <span className={props.prefixClasses} key="prefix">{props.prefix}</span>
     }
 
     if (props.suffix) {
         opts.suffix = React.isValidElement(props.suffix) ? props.suffix :
-            <><span className={props.suffixClasses}>{props.suffix}</span></>
+            <span className={props.suffixClasses} key="suffix">{props.suffix}</span>
     }
 
     opts.fn = function process(tag) {
         let tagLink = props.permalink
         tagLink = tagLink.replace(/:slug/, tag.slug) || `/${tag.slug}/`
 
-        return props.autolink ?
-            <>
```

Comment:
The `<>` as shorthand for `<React.Fragment>` are indeed not needed here, as the most outer html tag already functions as a wrapper. It's only needed when two html tags of the same level need to be wrapped.